students/FyfyNguyen/session03/mailroom.py

```diff
@@ -1,35 +1,103 @@
-#!/usr/bin python3
+#!/usr/bin/env python3
 
 """
-mailroom assignment
+Mailroom 1
 """
+import sys
+
+donors = [("Robyn Rihanna", [100, 200, 300]),
+          ("Ariana Grande", [2250, 4000, 1000]),
+          ("Beyonce Carter-Knowles", [150000, 3500, 25000]),
+          ("Aubrey Drake Graham", [15000, 5500.25, 1200]),
+          ("Justin Bieber", [2500, 250, 750.50])
+          ]
+
+
+def print_donors():
+    for donor in donors:
+        print(donor[0])
+
+
+def find_donor(name_entered):
+    for donor in donors:
+        if name_entered.strip().lower() == donor[0].lower():
+            return donor
+    return None
+
+
+def send_email(name_entered, donation):
+    print(
+        """Dear {},
+
+Your generous ${:,.2f} donation just made our day!
+
+Thank you!
+-The Charity""".format(name_entered.title(), donation)
+    )
 
-donors = [("Fred Jones", [100, 200, 300]),
-        ("Amy Schumer", [2000, 4000, 1000])
-        ]
 
 def thank_you():
-    print("do the thank you think now")
+    while True:
+        name_entered = input("Enter the name of a donor or 'list' to view "
+                             "donors >>> ").lower().strip()
+        if name_entered == "list":
+            print_donors()
+        else:
+            break
+
+    donation = float(input("Enter donation amount >>> "))
+    donor = find_donor(name_entered)
+    if donor is None:
+        donor = (name_entered.title(), [])
+        donor[1].append(donation)
```

Comment:
you have it exactly right: this is a mutable inside an immutable.

A tuple being immutable means you can't change its contents. -- that is list of donations has to be the SAME list -- but you can still change that list in place. what you can't do is replace it with a new list.